crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -202,7 +206,9 @@ impl DiskFileBlobStoreInner {
         {
             let _lock = self.file_lock.write();
             for (path, data) in raw {
-                if let Err(err) = fs::write(&path, &data) {
+                if path.exists() {
+                    debug!(target:"txpool::blob", ?path, "Blob already exists");
+                } else if let Err(err) = fs::write(&path, &data) {
```

Comment:
hmm, I think this is okay.

if we only need to check existence then, we could enforce this when opening the file and fail if it already exists via `OpenOptions::create_new`
but then we'd have to deal with error handling for this case, so this is okay.